### ISSUE TITLE
Change request - velocities for spacecraft-based observations

### ISSUE THREAD

**Issue by Bill-Gray** [IAU-ADES/ADES-Master#11]
At present,  the position of the spacecraft for WISE,  SOHO,  Gaia,  etc.-based observations can be provided in ADES,  via `pos1`, `pos2`,  and `pos3`.  You can even specify the upper triangular part of the covariance matrix via `posNN`,  1 <= N <= 3.  But you can't specify the spacecraft velocity.  Knowing this is quite useful when trying to diagnose timing errors.

I have "sort of" worked around this for my [on-line spacecraft offset adding tool](https://www.projectpluto.com/add_off.htm),  which only handles punched-card data,  by inserting `COM Spacecraft vel` lines before each pair of punch-card records for space-based observations;  for example,
```
COM Spacecraft vel 2.983150 -0.391510 -7.013880 km/s
     N00ktdv  S2022 10 31.01265 08 09 22.60 -09 35 29.0    ~B11I       LNEOCPC51
     N00ktdv  s2022 10 31.01265 1 - 5629.70018+ 2923.69205- 2564.71410  NEOCPC51
```
Software that "knows" about this can use the velocities,  and other software will just figure it's a comment.  For ADES,  I'd like to extend the "position" to be a "state vector".  We'd have `pos4`, `pos5`,  and `pos6`,  and the covariance matrix extended to be 6x6 with 21 elements in the upper triangle.  (With,  frequently,  the velocity bits left unset.)

I'd settle for `vel1`, `vel2`, `vel3`.  But realistically,  position and velocity are apt to be correlated and I'd like to see one 6x6 matrix instead of two 3x3 ones.

Thanks for considering this.

**Comment by stevechesley** [comment]
Bill, thanks for raising this. Spacecraft velocity information has been requested earlier to allow calculation of along- and cross-track residuals. It seems like a good time to go ahead with this, well before NEO Surveyor launches, and so I will add it to the to-do list. We also want to add the possibility of multi-band photometry that would be important for NEO Surveyor, and perhaps others.

We do want to avoid adding gratuitous fields, and so I'm wondering what the use case is for the velocity covariance, and especially the position and velocity correlation terms. I would expect that spacecraft velocity errors are going to be generally small enough (mm/s, say) that they won't affect questions about timing errors. Cross terms seem like they would be even less interesting. In contrast, the 3x3 position covariance has a big impact on how the astrometry should be processed, and so including that appears better justified.

**Comment by Bill-Gray** [comment]
Truthfully,  I don't have a use case in mind for the extended covariance matrix.  It seemed to me that if we're going to include the full state vector,  we should have the option of including the entire 6x6 covariance.  Some people might just report six elements from the 3x3;  others would report 21 from the 6x6.

Unless somebody comes up with a good reason otherwise,  I'd be happy to go along with just having the velocity data.  If we do come up with some oddball case requiring an "expanded" covariance,  we can do so then.

**Comment by stevechesley** [comment]
OK, so I'll put adding ```vel1```, ```vel2```, ```vel3``` on the TO-DO list and leave the issue open for now. Will consult with the MPC and we may do it at the same time we implement a solution to multi-band photometry  (Issue #2). That may be months away, but I don't think that any spacecraft pipelines are going to change their established pipelines to include this. But we need to be sure that it is all set long before NEO Surveyor.

**Comment by dfarnocchia** [comment]
The other discussion about Gaia-DR3 and gravitational light bending got me thinking.
If Gaia provides absolute measurement and not differential relative to stars, do we need to worry about relativistic aberration due to the spacecraft velocity relative to the solar system barycenter? If so the velocity is a requirement...
For traditional astrometry we don't need to worry about it because the effect is the same on stars and target and thus cancels out.
For Gaia-DR2, the astrometry was corrected for this effect: https://www.aanda.org/articles/aa/pdf/2018/08/aa32900-18.pdf
I would expect the same to happen to DR3.

**Comment by stevechesley** [comment]
Resolved with PR #47
